### PR TITLE
New Sync Options (collections and scrobbling)

### PR DESCRIPTION
**Pull request by hhergeth**
I added options to disable collection synchronization to Trakt as well as scrobbling. 
The main change was adding the two UI elements. The default values for both options are enabled so that the behavior is as it was before.

Other Information
- This is not the most efficient way to implement this as some work performed in SyncMovies and SyncShows is now useless.
- I could only test that it is working for my own account.

### CODE REVIEW

**Comment by joshuaboniface** [comment]
Looks good to me as long as it works! Are you able to clean up the useless functionality as well?

**Comment by hhergeth** [comment]
With my previous comment I meant that if the collection synchronization was turned off, some work was done pointlessly (compiling the list of what media is newly collected). I added checks to prevent that now.
Please let me know if there is anything else I should add/change.

**Comment by anthonylavado** [comment]
Ah okay, I understand the purpose of this now. Looks good to me.

File: Trakt/ScheduledTasks/SyncLibraryTask.cs [hunk 1]
```diff
@@ -208,7 +211,8 @@ public async Task Execute(CancellationToken cancellationToken, IProgress<double>
             }
 
             // send movies to mark collected
-            await SendMovieCollectionUpdates(true, traktUser, collectedMovies, progress.Split(4), cancellationToken).ConfigureAwait(false);
+            if(traktUser.SynchronizeCollections)
```
Comment (cvium) [inline Trakt/ScheduledTasks/SyncLibraryTask.cs]:
```suggestion
            if (traktUser.SynchronizeCollections)
```

File: Trakt/ScheduledTasks/SyncLibraryTask.cs [hunk 2]
```diff
@@ -208,7 +211,8 @@ public async Task Execute(CancellationToken cancellationToken, IProgress<double>
...
+                await SendMovieCollectionUpdates(true, traktUser, collectedMovies, progress.Split(4), cancellationToken).ConfigureAwait(false);
```
Comment (cvium) [inline Trakt/ScheduledTasks/SyncLibraryTask.cs]:
```suggestion
            {
                await SendMovieCollectionUpdates(true, traktUser, collectedMovies, progress.Split(4), cancellationToken).ConfigureAwait(false);
            }
```

File: Trakt/ScheduledTasks/SyncLibraryTask.cs [hunk 3]
```diff
@@ -363,19 +367,23 @@ public async Task Execute(CancellationToken cancellationToken, IProgress<double>
                     unplayedEpisodes.Add(episode);
                 }
 
-                var traktCollectedShow = SyncFromTraktTask.FindMatch(episode.Series, traktCollectedShows);
-                if (traktCollectedShow?.seasons == null
-                    || traktCollectedShow.seasons.All(x => x.number != episode.ParentIndexNumber)
-                    || traktCollectedShow.seasons.First(x => x.number == episode.ParentIndexNumber)
-                        .episodes.All(e => e.number != episode.IndexNumber))
+                if(traktUser.SynchronizeCollections)
```
Comment (cvium) [inline Trakt/ScheduledTasks/SyncLibraryTask.cs]:
```suggestion
                if (traktUser.SynchronizeCollections)
```

File: Trakt/ScheduledTasks/SyncLibraryTask.cs [hunk 4]
```diff
@@ -158,14 +158,17 @@ public async Task Execute(CancellationToken cancellationToken, IProgress<double>
                 var libraryMovie = child as Movie;
                 var userData = _userDataManager.GetUserData(user.Id, child);
 
-                // if movie is not collected, or (export media info setting is enabled and every collected matching movie has different metadata), collect it
-                var collectedMathingMovies = SyncFromTraktTask.FindMatches(libraryMovie, traktCollectedMovies).ToList();
-                if (!collectedMathingMovies.Any()
-                    || (traktUser.ExportMediaInfo
-                        && collectedMathingMovies.All(
-                            collectedMovie => collectedMovie.MetadataIsDifferent(libraryMovie))))
+                if(traktUser.SynchronizeCollections)
```
Comment (cvium) [inline Trakt/ScheduledTasks/SyncLibraryTask.cs]:
```suggestion
                if (traktUser.SynchronizeCollections)
```

File: Trakt/ScheduledTasks/SyncLibraryTask.cs [hunk 5]
```diff
@@ -363,19 +367,23 @@ public async Task Execute(CancellationToken cancellationToken, IProgress<double>
...
+                    {
+                        collectedEpisodes.Add(episode);
+                    }
                 }
 
                 decisionProgress.Report(100);
             }
 
-            await SendEpisodeCollectionUpdates(true, traktUser, collectedEpisodes, progress.Split(4), cancellationToken).ConfigureAwait(false);
+            if(traktUser.SynchronizeCollections)
```
Comment (cvium) [inline Trakt/ScheduledTasks/SyncLibraryTask.cs]:
```suggestion
            if (traktUser.SynchronizeCollections)
```

File: Trakt/ScheduledTasks/SyncLibraryTask.cs [hunk 6]
```diff
@@ -363,19 +367,23 @@ public async Task Execute(CancellationToken cancellationToken, IProgress<double>
...
+                await SendEpisodeCollectionUpdates(true, traktUser, collectedEpisodes, progress.Split(4), cancellationToken).ConfigureAwait(false);
```
Comment (cvium) [inline Trakt/ScheduledTasks/SyncLibraryTask.cs]:
```suggestion
            {
                await SendEpisodeCollectionUpdates(true, traktUser, collectedEpisodes, progress.Split(4), cancellationToken).ConfigureAwait(false);
            }
```

File: Trakt/ServerMediator.cs [hunk 7]
```diff
@@ -161,6 +161,9 @@ private async void KernelPlaybackStart(object sender, PlaybackProgressEventArgs
                     _logger.LogInformation("Could not match user with any stored credentials");
                     return;
                 }
+                
+                if (!traktUser.Scrobble)
+                    return;
```
Comment (cvium) [inline Trakt/ServerMediator.cs]:
```suggestion
                {
                    return;
                }
```

File: Trakt/ServerMediator.cs [hunk 8]
```diff
@@ -232,6 +235,9 @@ private async void KernelPlaybackStopped(object sender, PlaybackStopEventArgs e)
                     _logger.LogError("Could not match trakt user");
                     return;
                 }
+                
+                if (!traktUser.Scrobble)
+                    return;
```
Comment (cvium) [inline Trakt/ServerMediator.cs]:
```suggestion
                {
                    return;
                }
```

**Pull request merged**